src/Umbraco.Web/CompositionExtensions.cs

```diff
@@ -184,6 +184,30 @@ public static void SetSiteDomainHelper(this Composition composition, ISiteDomain
             composition.RegisterUnique(_ => helper);
         }
 
+        /// <summary>
+        /// Sets the default controller for rendering template views.
+        /// </summary>
+        /// <typeparam name="TController">The type of the controller.</typeparam>
+        /// <param name="composition">The composition.</param>
+        /// <remarks>The controller type is registered to the container by the composition.</remarks>
+        public static void SetDefaultRenderMvcController<TController>(this Composition composition)
+            => composition.SetDefaultRenderMvcController(typeof(TController));
+
+        /// <summary>
+        /// Sets the default controller for rendering template views.
+        /// </summary>
+        /// <param name="composition">The composition.</param>
+        /// <param name="controllerType">The type of the controller.</param>
+        /// <remarks>The controller type is registered to the container by the composition.</remarks>
+        public static void SetDefaultRenderMvcController(this Composition composition, Type controllerType)
```

Comment:
Is there any reason we would never know the type at compile time?

Reply:
yes, it is possible for ppl to replace the default controller - in case you want to ... change how all views are rendered without hijacking some content types, etc - so it's user-configurable